TodoApp/Src/TodoApp.Dependency/Exceptions/DependencyResolutionFailedException.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+
+namespace TodoApp.Dependency.Exceptions
+{
+    internal class DependencyResolutionFailedException : Exception
+    {
+        public DependencyResolutionFailedException()
+        {
+        }
+
+        public DependencyResolutionFailedException(string message) 
```

Comment:
It is perfectly OK to omit constructor you do not need - such as parameterless and message-only ones. However, have a look at his [`SO`](https://stackoverflow.com/a/1066714/1138663) – making the exception serializable should be really trivial in your case and it makes it much more robust (making them serializable is actually a nice habit for anybody to form).